test/1_stdlib/SetAlgebra.swift

```diff
@@ -0,0 +1,188 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+public struct DefaultedSetAlgebra<Element> : SetAlgebra {
+  public typealias Element = String
+  var store = Set<String>()
+  public init(){}
+  public init(_ elements: [Element]) {
+    for el in elements {
+      store.insert(el)
+    }
+  }
+
+  public func contains(_ member: String) -> Bool {
+    return store.contains(member)
+  }
+
+  public func union(_ other: DefaultedSetAlgebra) -> DefaultedSetAlgebra {
+    var defaultedSetAlgebra = DefaultedSetAlgebra()
+    defaultedSetAlgebra.store = store.union(other.store)
+    return defaultedSetAlgebra
+  }
+
+  public func intersection(_ other: DefaultedSetAlgebra) -> DefaultedSetAlgebra {
+    var defaultedSetAlgebra = DefaultedSetAlgebra()
+    defaultedSetAlgebra.store = store.intersection(other.store)
+    return defaultedSetAlgebra
+  }
+
+  public func symmetricDifference(_ other: DefaultedSetAlgebra) -> DefaultedSetAlgebra {
+    var defaultedSetAlgebra = DefaultedSetAlgebra()
+    defaultedSetAlgebra.store = store.symmetricDifference(other.store)
+    return defaultedSetAlgebra
+  }
+
+  public mutating func insert(_ newMember: String) -> (inserted: Bool, memberAfterInsert: String) {
+    return store.insert(newMember)
+  }
+
+  public mutating func remove(_ member: String) -> String? {
+    return store.remove(member)
+  }
+
+  public mutating func update(with newMember: String) -> String? {
+    return store.update(with: newMember)
+  }
+
+  public mutating func formUnion(_ other: DefaultedSetAlgebra) {
+    store.formUnion(other.store)
+  }
+
+  public mutating func formSymmetricDifference(_ other: DefaultedSetAlgebra) {
+    store.formSymmetricDifference(other.store)
+  }
+
+  public mutating func formIntersection(_ other: DefaultedSetAlgebra) {
+    store.formIntersection(other.store)
+  }
+}
+
+public func ==<Element>(lhs: DefaultedSetAlgebra<Element>, rhs: DefaultedSetAlgebra<Element>) -> Bool {
+  return lhs.store == rhs.store
+}
+
+let SetAlgebraTests = TestSuite("SetAlgebra")
+
+SetAlgebraTests.test("contains") {
```

Comment:
Instead of testing with strings, which can have some strange comparison semantics at times (and just in general pull in a lot of Unicode machinery), we use `MinimalHashableValue`.  Could you change the tests to use it instead?